src/nvim/edit.c

```diff
@@ -150,6 +154,22 @@ struct compl_S {
   int cp_number;                // sequence number
 };
 
+// Structure that stores the score of each completion entry
+// Used to sort the pum items.
+typedef struct compl_score_S compl_score_T;
+struct compl_score_S {
+  compl_T   *compl;
+  double score;
+};
+
+// lua functions used to filter completion matches
+// global_filterfunc is used for all completion sources,
+// while user_filterfunc is only used to filter matches
+// from vim.api.nvim_complete().
+LuaRef *global_filterfunc = NULL;
```

Comment:
declare these in `edit.h` instead as `EXTERN LuaRef global_filterfunc INIT(= NO_REF)`, then it is initialized once but available by including `edit.h` anywhere.

Reply:
This is helpful, thanks. I'm still very new to C, so I'm not very knowledgeable about things like this.